Sitemap.php

```diff
@@ -173,66 +204,30 @@ public function write()
 
     /**
      * Flushes buffer into file
-     * @param bool $finishFile Pass true to close the file to write to, used only when useGzip is true
+     *
+     * @param int $footSize Size of the remaining closing tags
+     * @throws \OverflowException
      */
-    private function flush($finishFile = false)
+    private function flush($footSize = 10)
     {
-        if ($this->useGzip) {
-            $this->flushGzip($finishFile);
-            return;
-        }
-        file_put_contents($this->getCurrentFilePath(), $this->writer->flush(true), FILE_APPEND);
-    }
-
-    /**
-     * Decides how to flush buffer into compressed file
-     * @param bool $finishFile Pass true to close the file to write to
-     */
-    private function flushGzip($finishFile = false) {
-        if (function_exists('deflate_init') && function_exists('deflate_add')) {
-            $this->flushWithIncrementalDeflate($finishFile);
-            return;
-        }
-        $this->flushWithTempFileFallback($finishFile);
-    }
-
-    /**
-     * Flushes buffer into file with incremental deflating data, available in php 7.0+
-     * @param bool $finishFile Pass true to write last chunk with closing headers
-     */
-    private function flushWithIncrementalDeflate($finishFile = false) {
-        $flushMode = $finishFile ? ZLIB_FINISH : ZLIB_NO_FLUSH;
-
-        if (empty($this->deflateContext)) {
-            $this->deflateContext = deflate_init(ZLIB_ENCODING_GZIP);
-        }
-        
-        $compressedChunk = deflate_add($this->deflateContext, $this->writer->flush(true), $flushMode);
-        file_put_contents($this->getCurrentFilePath(), $compressedChunk, FILE_APPEND);
-
-        if ($finishFile) {
-            $this->deflateContext = null;
-        }
-    }
-
-    /**
-     * Flushes buffer into temporary stream and compresses stream into a file on finish
-     * @param bool $finishFile Pass true to compress temporary stream into desired file
-     */
-    private function flushWithTempFileFallback($finishFile = false) {
-        if (empty($this->tempFile) || !is_resource($this->tempFile)) {
-            $this->tempFile = fopen('php://temp/', 'w');
+        $data = $this->writer->flush(true);
+
+        /**
+         * Limit the file size of each single site map
+         *
+         * We use a heuristic of 10 Bytes for the remainder of the file,
+         * i.e. </urlset> plus a new line.
+         */
+        if ($this->byteCount + strlen($data) + $footSize > $this->maxBytes) {
```

Comment:
`mb_strlen($data, '8bit');` should be used in order not to break when mb_ overload is active in php.ini.

Reply:
Ah thx, I did not know about that.